lib/compiler.js

```diff
@@ -54,36 +54,36 @@ function scan (text) {
 }
 
 function parse (tokens) {
-  return _buildTree(tokens, {}, [])
+  return _buildTree(tokens, {}, [], [])
 }
 
-function _buildTree (tokens, parent, stack) {
+function _buildTree (tokens, parent, arrayStack, objectStack) {
   var props = {}
   var token
   var peek
 
   while ((token = tokens.shift())) {
     if (token.tag === '_n') {
       token.type = 'object'
-      token.properties = _buildTree(tokens, token, stack)
+      token.properties = _buildTree(tokens, token, arrayStack, objectStack)
       // exit if in object stack
-      peek = stack[stack.length - 1]
+      peek = objectStack[objectStack.length - 1]
       if (peek && (peek.tag === '/')) {
-        stack.pop()
+        objectStack.pop()
         _addToken(token, props)
         return props
       }
     } else if (token.tag === ',') {
       return props
     } else if (token.tag === '(') {
-      stack.push(token)
+      arrayStack.push(token)
       parent.type = 'array'
       continue
     } else if (token.tag === ')') {
-      stack.pop(token)
+      arrayStack.pop(token)
```

Comment:
> And then I think I'd use slice instead of splice since the former takes an index and the latter a count.

I am not quite sure what you mean here. If the stack looks like this `['(', '(', '/', '/']` and the current token is `)`, we need to "pop" `tagStack[1]`. Using slice will look like this 
```javascript
stack = stack.slice(0, index).concat(stack.slice(index+1, stack.length)
```